plugin/driver/network_utils.go

```diff
@@ -134,3 +134,123 @@ func getGratuitousArp(mac net.HardwareAddr, ip net.IP) []byte {
 
 	return buf.Bytes()
 }
+
+type NetworkIfc struct {
+	ifc_name string
+	mac_addr string
+	ip       string
+	mask     string
+}
+
+func GetInterfaceInformation(net_ns_path string) ([]NetworkIfc, error) {
+
+	var netns ns.NetNS
+	var err error
+
+	netns, err = ns.GetNS(net_ns_path)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open netns %s: %v", net_ns_path, err)
+	}
+	defer netns.Close()
+
+	var net_info []NetworkIfc
+
+	// Let's go in
+	err = netns.Do(func(_ ns.NetNS) error {
+		// Get Ifcs on container
+		ifc_list, err := net.Interfaces()
+
+		if err != nil {
+			return fmt.Errorf("Failed to get interfaces in Netns: %v", err)
+		}
+
+		// For some reason I can't properly copy the full interface info struct outside. Seems the go net library is doing some
+		// caching inbetween (checked on source code)
+
+		// This is pathetic, but until I have a way to copy the full struct and substructs, I have to do this.
+		// Quite the chance the return is not static data, but instead is computed every time yout call, hence data changes between
+		// invocations in and out of the netNS
+		for _, ifc := range ifc_list {
```

Comment:
This is redundant, as we are doing the same in RunContainerArping where we enter the namespace and fetch the address to generate the ARP. I would suggest rather having two separate go routines accessing information separately should be unified.